src/index.ts

```diff
@@ -0,0 +1,169 @@
+        }
+        res
+          .writeHead(200, "Ok", {
+            "Content-Type": "text/json",
+          })
+          .end(content);
+      });
+      break;
+    case "/post":
+      res.setHeader("content-type", "text/plain");
+
+      if (req.method !== "POST") {
+        return res.writeHead(403, "Forbidden").end(`Error 403. Forbidden`);
+      }
+
+      let body = "";
+      req.on("data", (chunk) => {
+        body += chunk.toString();
+      });
+
+      req.on("end", () => {
+        const parsedBody: ParsedQuery<string> = parse(body);
+        if (!parsedBody.name || !parsedBody.age) {
+          return res
+            .writeHead(400, "Bad Request")
+            .end(`Error 400. Bad Request`);
+        }
+
+        fs.readFile(pathToJsonFile, "utf-8", (err, content) => {
+          if (err) {
+            return res
+              .writeHead(500, "Internal Server Error")
+              .end(`Error 500. Internal Server Error`);
+          }
+
+          const users: UsersType = JSON.parse(content);
+          const newUser: typeOfUser = {
+            id: getNextId(users),
+            name: parsedBody.name?.toString() || "",
+            age: parsedBody.age?.toString() || "",
+          };
+          users.push(newUser);
+
+          fs.writeFile(pathToJsonFile, JSON.stringify(users), () => {
+            return res
+              .writeHead(201, "Created")
+              .end(`Data has been successfully written to file`);
+          });
+        });
+      });
+      break;
+    case "/put":
+      res.setHeader("content-type", "text/plain");
+
+      if (req.method !== "PUT") {
+        return res.writeHead(403, "Forbidden").end(`Error 403. Forbidden`);
+      }
+
+      let bodyOfPut = "";
+      req.on("data", (chunk) => {
+        bodyOfPut += chunk.toString();
+      });
+
+      req.on("end", () => {
+        const parsedBody: ParsedQuery<string> = parse(bodyOfPut);
+
+        if (!parsedBody.id || !parsedBody.name || !parsedBody.age) {
+          return res
+            .writeHead(400, "Bad Request")
+            .end(`Error 400. Bad Request`);
+        }
+
+        fs.readFile(pathToJsonFile, "utf-8", (err, content) => {
+          if (err) {
+            return res
+              .writeHead(500, "Internal Server Error")
+              .end(`Error 500. Internal Server Error`);
+          }
+
+          const users: UsersType = JSON.parse(content);
+
+          const changedUser: typeOfUser = {
+            id: Number(parsedBody.id) || 0,
+            name: parsedBody.name?.toString() || "",
+            age: parsedBody.age?.toString() || "",
+          };
+          const newListOfUsers = users.map((user) => {
+            if (user.id === changedUser.id) {
+              return changedUser;
+            }
+            return user;
+          });
+
+          fs.writeFile(pathToJsonFile, JSON.stringify(newListOfUsers), () => {
+            return res
+              .writeHead(201, "Created")
```

Comment:
201 http status code is not for PUT endpoint.
For Put endpoint, file should be already exists and you just rewrite content.

Reply:
Changed